aws/resource_aws_macie_s3_bucket_association_test.go

```diff
@@ -0,0 +1,174 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/macie"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSMacieS3BucketAssociation_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSMacieS3BucketAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSMacieS3BucketAssociationConfig_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMacieS3BucketAssociationExists("aws_macie_s3_bucket_association.test"),
+					resource.TestCheckResourceAttr("aws_macie_s3_bucket_association.test", "classification_type.0.one_time", "false"),
+				),
+			},
+			{
+				Config: testAccAWSMacieS3BucketAssociationConfig_basicOneTime(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMacieS3BucketAssociationExists("aws_macie_s3_bucket_association.test"),
+					resource.TestCheckResourceAttr("aws_macie_s3_bucket_association.test", "classification_type.0.one_time", "true"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSMacieS3BucketAssociation_accountIdAndPrefix(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSMacieS3BucketAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSMacieS3BucketAssociationConfig_accountIdAndPrefix(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMacieS3BucketAssociationExists("aws_macie_s3_bucket_association.test"),
+					resource.TestCheckResourceAttr("aws_macie_s3_bucket_association.test", "classification_type.0.one_time", "false"),
+				),
+			},
+			{
+				Config: testAccAWSMacieS3BucketAssociationConfig_accountIdAndPrefixOneTime(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMacieS3BucketAssociationExists("aws_macie_s3_bucket_association.test"),
+					resource.TestCheckResourceAttr("aws_macie_s3_bucket_association.test", "classification_type.0.one_time", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSMacieS3BucketAssociationDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).macieconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_macie_s3_bucket_association" {
+			continue
+		}
+
+		req := &macie.ListS3ResourcesInput{}
+		acctId := rs.Primary.Attributes["member_account_id"]
+		if acctId != "" {
+			req.MemberAccountId = aws.String(acctId)
+		}
+
+		for {
+			resp, err := conn.ListS3Resources(req)
```

Comment:
We should prefer to also use the paginated method here 👍

Reply:
👍